### PR TITLE
Support Azure log credentials in `update_external_connection`

### PR DESCRIPTION
**Pull request by Neon-White**
### Explain the changes
1. Added support for the optional Azure log credentials in `update_external_connection`
2. Fixed a bug in the Azure log iterator

### CODE REVIEW

File: src/server/system_services/account_server.js [hunk 1]
```diff
@@ -793,6 +793,7 @@ async function update_external_connection(req) {
             endpoint: connection.endpoint,
             cp_code: connection.cp_code,
             auth_method: connection.auth_method,
+            azure_log_access_keys: connection.azure_log_access_keys,
```
Comment (romayalon) [inline src/server/system_services/account_server.js]: I'll move the [old PR discussion](https://github.com/noobaa/noobaa-core/pull/7308#discussion_r1258238501) to here, I def agree with not printing secrets (get_account_info()), but we do return identity so we also need to return the ids from the azure_log_access_keys object

```
azure_log_access_keys: {
            properties: {
                azure_tenant_id -> should be printed to the user
                azure_client_id -> should be printed to the user
                azure_client_secret ,
                azure_logs_analytics_workspace_id  -> should be printed to the user
            }
 }
```

what's the point of letting the user update these properties without him knowing what he put there in the first place

Comment (dannyzaken) [reply src/server/system_services/account_server.js]: @Neon-White do we want to treat all of these the same as secret_key, or does any of these fields should be returned to identify the credentials? 
e.g., for azure connections, we use the Account Name as the "identity" (`AWS_ACCESS_KEY_ID` for S3 connections). should any of these new fields be returned as part of the account information?

File: src/server/system_services/account_server.js [hunk 2]
```diff
@@ -791,6 +790,7 @@ async function update_external_connection(req) {
             endpoint: connection.endpoint,
             cp_code: connection.cp_code,
             auth_method: connection.auth_method,
+            azure_log_access_keys: azure_log_access_keys || undefined,
```
Comment (romayalon) [inline src/server/system_services/account_server.js:793]: `|| undefined` is redundant, no?

File: src/server/system_services/account_server.js [hunk 3]
```diff
@@ -803,15 +803,34 @@ async function update_external_connection(req) {
         throw new RpcError('INVALID_CREDENTIALS', `Credentials are not valid ${name}`);
     }
 
+    let set_obj;
+    if (azure_log_access_keys?.azure_tenant_id?.unwrap()) {
```
Comment (romayalon) [inline src/server/system_services/account_server.js:807]: why checking for azure_tenant_id existence and not for azure_log_access_keys?

Comment (romayalon) [inline src/server/system_services/account_server.js:807]: having Azure log access keys is a very small scenario and I think that the code flow should be:
1. create set_obj // -> main flow first
2. if (azure_log_access_keys) set the azure log fields on set_obj // -> other flows

Comment (romayalon) [reply src/server/system_services/account_server.js:811]: try adding +optional to the operator code

Comment (dannyzaken) [reply src/server/system_services/account_server.js:811]: I think that here we should validate the values of `azure_log_access_keys`. we already have the [schema](https://github.com/noobaa/noobaa-core/blob/845073892c4768ec6e0edf593b0baadd20686a85/src/api/common_api.js#L754C1-L754C1) that enforces all the required fields. if these fields are invalid we should throw an error.
in the operator we should make sure we call this API correctly

**Review by romayalon — APPROVED** [review]

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**